upup/pkg/fi/cloudup/awsup/machine_types.go

```diff
@@ -58,2503 +60,37 @@ func (m *AWSMachineTypeInfo) EphemeralDevices() []*EphemeralDevice {
 	return disks
 }
 
-func GetMachineTypeInfo(machineType string) (*AWSMachineTypeInfo, error) {
-	for i := range MachineTypes {
-		m := &MachineTypes[i]
-		if m.Name == machineType {
-			return m, nil
+func GetMachineTypeInfo(c AWSCloud, machineType string) (*AWSMachineTypeInfo, error) {
+
+	info, err := c.DescribeInstanceType(machineType)
+	if err != nil {
+		return nil, err
+	}
+	machine := AWSMachineTypeInfo{
+		Name:              machineType,
+		GPU:               info.GpuInfo != nil,
+		InstanceENIs:      intValue(info.NetworkInfo.MaximumNetworkInterfaces),
+		InstanceIPsPerENI: intValue(info.NetworkInfo.Ipv4AddressesPerInterface),
+	}
+	memoryGB := float64(intValue(info.MemoryInfo.SizeInMiB)) / 1024
+	machine.MemoryGB = float32(math.Round(memoryGB*100) / 100)
```

Comment:
Nit: Should we remove the rounding logic once we're happy we get matching values?  Maybe just add a todo?